base/experimental.jl

```diff
@@ -129,6 +129,13 @@ Set the maximum number of potentially-matching methods considered when running i
 for methods defined in the current module. This setting affects inference of calls with
 incomplete knowledge of the argument types.
 
+The benefit of this setting is to reduce compilation in some cases. For example, when
+`@max_methods 2` is set and there are two potentially-matching methods returning
+different types inside a function body, then Julia will compile subsequent calls for
+both types so that they can be linked to inside the compiled function body. This
+speculative compilation can be avoided by setting `@max_methods 1`. Then, the
+compiled function will do a runtime dispatch instead.
```

Comment:
```suggestion
The benefit of this setting is to avoid excessive compilation and reduce invalidation risks
in poorly-inferred cases. For example, when `@max_methods 2` is set and there are two 
potentially-matching methods returning different types inside a function body, then Julia 
will compile subsequent calls for both types so that the compiled function body accounts for
both possibilities, and also the compiled code is vulnerable to invalidations that would 
happen when either of the two methods gets invalidated. This speculative compilation and
invalidation risk can be avoided by setting `@max_methods 1` and allowing the compiled code
to resort to runtime dispatch instead.
```

Reply:
Thanks! Nice improvement. I've manually took over your suggestion to get rid of some trailing whitespaces and I've also made some minor tweaks for readability